test/publish-agent.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+set -eo pipefail
+
+cat << EOF
+
+Test publishing end-to-end (agent tarballs):
+ * uses real schema from URL
+ * publishes into testing S3 bucket
+ * asserts assets were published on bucket
+EOF
+
+# Public (forwarded) env-vars
+TAG=1.16.3
+AWS_REGION="us-east-1"
+AWS_S3_BUCKET_NAME="nr-downloads-ohai-testing"
+AWS_S3_LOCK_BUCKET_NAME="onhost-ci-lock-testing"
+AWS_ROLE_SESSION_NAME="caos_testing"
+AWS_ROLE_ARN="arn:aws:iam::017663287629:role/caos_testing"
+DEST_PREFIX="infrastructure_agent/test_e2e/$(uuidgen)"
+# Private vars
+_ASSERT_DIR="s3://${AWS_S3_BUCKET_NAME}/${DEST_PREFIX}binaries/linux/386/"
+_ASSERT_FILE="newrelic-infra_linux_${TAG}"
```

Comment:
probably we also need `.tar.gz` here

Reply:
actually s3 ls will return all files matching the prefix and exit with 0 code if there's a match

ie:
```
> aws s3 ls s3://nr-downloads-main/infrastructure_agent/linux/yum/el/7/x86_64/newrelic-infra-1.16.0-1.el7.x86_64.rpm
2021-03-24 11:41:56   41318138 newrelic-infra-1.16.0-1.el7.x86_64.rpm
> echo $?                                                                                                                  
0
> aws s3 ls s3://nr-downloads-main/infrastructure_agent/linux/yum/el/7/x86_64/newrelic-infra-1.16.0
2021-03-24 11:41:56   41318138 newrelic-infra-1.16.0-1.el7.x86_64.rpm
> echo $?
0
```

Reply:
adding comment to clarify

Reply:
I didn't know, thank you!